apps/webservice/src/app/[workspaceSlug]/systems/[systemSlug]/environments/SidebarEnvironmentPanel.tsx

```diff
@@ -184,21 +198,27 @@ export const SidebarEnvironmentPanel: React.FC = () => {
               )
             </span>
             {node.data.targetFilter != null && (
-              <Link
-                href={`/${workspaceSlug}/targets?${new URLSearchParams({
-                  filter: LZString.compressToEncodedURIComponent(
-                    JSON.stringify(node.data.targetFilter),
-                  ),
-                })}`}
-                passHref
-              >
-                <Button variant="ghost" size="sm" asChild>
-                  <a target="_blank" rel="noopener noreferrer">
-                    <IconExternalLink className="mr-1 h-4 w-4" />
-                    View Targets
-                  </a>
-                </Button>
-              </Link>
+              <>
+                <TargetFilterUniquenessIndicator
+                  nodes={nodes}
+                  workspaceId={workspace.data?.id ?? ""}
+                />
+                <Link
+                  href={`/${workspaceSlug}/targets?${new URLSearchParams({
+                    filter: LZString.compressToEncodedURIComponent(
+                      JSON.stringify(node.data.targetFilter),
+                    ),
+                  })}`}
+                  passHref
+                >
+                  <Button variant="ghost" size="sm" asChild>
+                    <a target="_blank" rel="noopener noreferrer">
```

Comment:
? Link in a button in a link? This doesn't seem right

Reply:
I would remove the button as use the buttonVariantCn function or whatever its called, and just have the root link

Reply:
Sounds good. This is better.